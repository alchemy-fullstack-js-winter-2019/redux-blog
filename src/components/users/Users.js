import React from 'react';
import PropTypes from 'prop-types';

function Users({users}) {
  const listOfUsers = users.map(user => <li key={user.id}>{user}</li>);

  return (
    <>
    <ul>
      {listOfUsers}
    </ul>
    </>
  )
}

Users.propTypes = {
  users: PropTypes.array.isRequired
};

export default Users;
