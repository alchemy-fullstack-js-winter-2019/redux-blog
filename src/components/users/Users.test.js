import React from 'react';
import renderer from 'react-test-renderer';
import Users from './Users';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Posts', () => {
  it('matches a snapshot', () => {
    const users = [{
      name: 'name',
      username: 'username',
      email: 'email@email'
    }];
    const tree = renderer.create(
      <Router>
        <Users users={users} />
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
