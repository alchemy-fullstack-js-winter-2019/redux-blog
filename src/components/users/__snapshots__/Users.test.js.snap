// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts matches a snapshot 1`] = `
Array [
  <h2
    className="usersTitle"
  >
    Users
  </h2>,
  <ol>
    <li
      className="userLink"
    >
      <div>
        <div
          className="MuiGrid-item-5 MuiGrid-grid-xs-12-44 MuiGrid-grid-md-6-66"
        >
          <h6
            className="MuiTypography-root-101 MuiTypography-h6-118"
          />
          <div>
            <ul
              className="MuiList-root-137 MuiList-padding-138"
            >
              <li
                className="MuiListItem-root-141 MuiListItem-default-144 MuiListItem-dense-145 MuiListItem-gutters-149"
                disabled={false}
              >
                <div
                  className="MuiAvatar-root-156 MuiAvatar-colorDefault-157"
                >
                  <img
                    src="../../../account.png"
                  />
                </div>
                <a
                  className="link"
                  href="/users/undefined"
                  onClick={[Function]}
                >
                  <div
                    className="MuiListItemText-root-159"
                  />
                  <h2>
                    name
                  </h2>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </li>
  </ol>,
]
`;
